rocketpy/mathutils/function.py

```diff
@@ -2998,6 +2953,8 @@ def savetxt(
             file.write(header_line + newline)
             np.savetxt(file, data_points, fmt=fmt, delimiter=delimiter, newline=newline)
 
+    # Input validators
+
     @staticmethod
     def _check_user_input(
         source,
```

Comment:
So, `_check_user_input` is really only validating `source`, but it also calls other validation methods inside it.

I really dislike this, the way this validation is structured is really confusing. Here is a suggestion:

1. In `__init__` we call `set_source`, `set_inputs`, `set_interpolation`, `set_extrapolation` and `set_title`.
2. Inside each of these methods, the respective validation method is called, e.g. in `set_source` call `_validate_source`

Currently it is very confusing to understand what is being validated and when.

Reply:
Im working on this. Should open another PR soon

Reply:
https://github.com/RocketPy-Team/RocketPy/pull/582